webrender/src/prim_store.rs

```diff
@@ -677,23 +662,55 @@ impl PrimitiveStore {
                                    resource_cache: &mut ResourceCache,
                                    frame_id: FrameId,
                                    device_pixel_ratio: f32,
+                                   dummy_mask_cache_item: &TextureCacheItem,
                                    auxiliary_lists: &AuxiliaryLists) -> bool {
         let metadata = &mut self.cpu_metadata[prim_index.0];
         debug_assert!(metadata.need_to_build_cache);
         metadata.need_to_build_cache = false;
 
-        if let Some(mask_key) = metadata.mask_image {
-            let tex_cache = resource_cache.get_image(mask_key, ImageRendering::Auto, frame_id);
+        let clip_data_mask = match metadata.clip_info.as_ref() {
```

Comment:
Does this mean that we'll end up with a complex clip region enabled when the clip region is a simple rect with zero radius? If so, that will introduce a lot more expensive clip shaders than needed.